internal/pkg/ociimage/unpack.go

```diff
@@ -88,15 +85,26 @@ func UnpackRootfs(_ context.Context, srcImage v1.Image, destDir string) (err err
 		mapOptions.GIDMappings = append(mapOptions.GIDMappings, gidMap)
 	}
 
-	// Unpack root filesystem
-	unpackOptions := umocilayer.UnpackOptions{MapOptions: mapOptions}
-	err = umocilayer.UnpackLayer(destDir, flatTar, &unpackOptions)
-	if err != nil {
-		return fmt.Errorf("error unpacking rootfs: %s", err)
+	for _, l := range layers {
+		layerDigest, err := l.Digest()
+		if err != nil {
+			return fmt.Errorf("while getting digest: %w", err)
+		}
+		sylog.Debugf("Extracting layer %s", layerDigest)
+		layerReader, err := l.Uncompressed()
+		if err != nil {
+			return fmt.Errorf("while reading layer: %s: %w", layerDigest, err)
+		}
+		unpackOptions := umocilayer.UnpackOptions{MapOptions: mapOptions}
+		err = umocilayer.UnpackLayer(destDir, layerReader, &unpackOptions)
+		if err != nil {
+			return fmt.Errorf("while unpacking layer %s: %w", layerDigest, err)
+		}
+		if err := layerReader.Close(); err != nil {
+			return fmt.Errorf("while closing layer %s: %w", layerDigest, err)
+		}
```

Comment:
This should happen even if `umocilayer.UnpackLayer` fails? Might be worth a `defer` after `l.Uncompressed`?

Reply:
I'll do it explictly in the error case, rather than defer, if that's okay. That avoids having 74 defers that do nothing stacked up for the large image :-)